### PR TITLE
chore: added the used code from mendersoftware/mender

### PR DESCRIPTION
**Pull request by danielskinstad**
Mender-snapshot used github.com/mendersoftware/mender as a go module. The repo is not going to be updated further, and little code is used. Added the used code to this code base.

Changelog: title

Ticket: MEN-6896

### CODE REVIEW

File: system/ioctl.go [hunk 1]
```diff
@@ -27,22 +26,17 @@ import (
...
-	ErrDevNotMounted = fmt.Errorf("device not mounted")
-	NotABlockDevice  = fmt.Errorf("not a block device")
-)
-
-// MountInfo maps a single line in /proc/<pid|self>/mountinfo
-// See the linux kernel documentation: linux/Documentation/filesystems/proc.txt
-// A line takes the form:
-// 36 35 98:0 /mnt1 /mnt2 rw,noatime master:1 - ext3 /dev/root rw,errors=continue
-// (1)(2)(3)   (4)   (5)      (6)      (7)   (8) (9)   (10)         (11)
 type MountInfo struct {
```
Comment (lluiscampos) [inline system/ioctl.go]: This comment seems useful. I would restore it

File: system/ioctl.go [hunk 2]
```diff
@@ -240,65 +195,39 @@ func getUbiDeviceSize(file *os.File) (uint64, error) {
...
+	case unix.S_IFDIR,
+		unix.S_IFREG,
+		unix.S_IFLNK:
+		return [2]uint32{
+			unix.Major(uint64(stat.Dev)), unix.Minor(uint64(stat.Dev))}, nil
 	}
-	return nil
+	return [2]uint32{^uint32(0), ^uint32(0)},
+		fmt.Errorf("invalid stat(2) st_mode %04X", devType)
 }
```
Comment (lluiscampos) [inline system/ioctl.go:236]: This file is really hard to review. As we have moved the functions around the diff shows many functions are removed and added. I looked and some of the functions to see that there were no code differences, so I assume that none of them had code differences :)

File: go.sum [hunk 3]
```diff
@@ -1,1022 +1,43 @@
...
-honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
-honnef.co/go/tools v0.0.0-20190106161140-3f1c8253044a/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
-honnef.co/go/tools v0.0.0-20190418001031-e561f6794a2a/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
-honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
-honnef.co/go/tools v0.0.1-2019.2.3/go.mod h1:a3bituU0lyd329TUQxRnasdCoJDkEUEAqEt0JzvZhAg=
-honnef.co/go/tools v0.0.1-2020.1.3/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9vFzvIQ3k=
-honnef.co/go/tools v0.0.1-2020.1.4/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9vFzvIQ3k=
-rsc.io/binaryregexp v0.2.0/go.mod h1:qTv7/COck+e2FymRvadv62gMdZztPaShugOCi3I+8D8=
-rsc.io/quote/v3 v3.1.0/go.mod h1:yEA65RcK8LyAZtP9Kv3t0HmxON59tX3rD+tICJqUlj0=
-rsc.io/sampler v1.3.0/go.mod h1:T1hPZKmBbMNahiBKFy5HrXp6adAjACjK9JXDnKaTXpA=
```
Comment (lluiscampos) [inline go.sum:1022]: I re-generated this locally on my machine and it looks good :+1:

**Review by lluiscampos — CHANGES_REQUESTED** [review]
The code looks good, nice clean-up!

This commit is not a feature, so we should not mark it as such. Use `chore` better. Read [here](https://github.com/mendersoftware/mendertesting/blob/master/commitlint/grammar.md) to get more familiar with the categories that we use for our commits.

There are several failures in CI that you need to amend, like licenses or static checks. Let me know if you want assistance to fix them!
:point_right:  From the CI checks, one that was strange to me is that CI did not validate the signature of your commit because it has an extra space. Did you use the `-s` flag of `git commit` or did you type that yourself?

**Review by lluiscampos — APPROVED** [review]
> This commit is not a feature, so we should not mark it as such. Use `chore` better. Read [here](https://github.com/mendersoftware/mendertesting/blob/master/commitlint/grammar.md) to get more familiar with the categories that we use for our commits.

I forgot to add that we should not have a changelog entry for this change neither. It does not add functionality nor fixes any bug. You can just remove the `changelog` line in your commit.

Otherwise looks good, pre-aproving.

Apart from fixing the commit, we need to run the integration pipeline :robot:

**Comment by danielskinstad** [comment]
@mender-test-bot start pipeline

**Pull request merged**